### PR TITLE
fix gts ts imports issue

### PR DESCRIPTION
**Pull request by patricklx**
run the processing in pre to ensure we have the imports before other plugin runs.

fixes https://github.com/embroider-build/ember-auto-import/issues/611

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -194,6 +174,15 @@ export function makePlugin<EnvSpecificOptions>(loadOptions: (opts: EnvSpecificOp
           },
         },
 
+        Identifier(path: NodePath<t.Identifier>) {
```
Comment (patricklx) [inline src/plugin.ts:177]: this ensures that newly added identifiers do actually reference the import binding. Otherwise ts plugin will remove the import

**Comment by ef4** [comment]
Can we add a test that runs our plugin and babel-plugin-transform-typescript together so that we don't regress on this bug?

**Comment by patricklx** [comment]
> Can we add a test that runs our plugin and babel-plugin-transform-typescript together so that we don't regress on this bug?

We have it already

**Comment by NullVoxPopuli** [comment]
> We have it already

it wasn't failing? :sweat_smile:

**Comment by patricklx** [comment]
> > We have it already
> 
> it wasn't failing? 😅

No, because it's not a problem of ts plugin +  this plugin. But a problem of
ts plugin + other plugin+ this plugin.

Updated the description

**Comment by ef4** [comment]
Let me rephrase my question: can we add a test they would fail without these changes?

**Comment by patricklx** [comment]
@ef4 added a test

File: src/plugin.ts (hunk 1 above)
Comment (patricklx) [reply src/plugin.ts:177]: this identifiers are also added by babel import util

Comment (patricklx) [reply src/plugin.ts:177]: This is only a problem because we do this in pre

Comment (patricklx) [reply src/plugin.ts:177]: This is running in pre as well. So it only affects identifiers added during ast transform

Comment (patricklx) [reply src/plugin.ts:177]: i could also have my own custom plugins that do some babel ast changes during template transform

**Comment by ef4** [comment]
Thanks, I found a way to use babel's `scope.crawl()` to update bindings for only the sub-trees of the AST that we're altering.
